OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -163,6 +169,20 @@ public Aircraft(ActorInitializer init, AircraftInfo info)
 			IsPlane = !info.CanHover;
 		}
 
+		public virtual IEnumerable<VariableObserver> GetVariableObservers()
+		{
+			if (Info.LandOnCondition != null)
+				yield return new VariableObserver(LandIfConditionChanged, Info.LandOnCondition.Variables);
+		}
+
+		void LandIfConditionChanged(Actor self, IReadOnlyDictionary<string, int> conditions)
```

Comment:
Just a couple of nits here:

1. Rename `LandIfConditionChanged`; the current wording means that it should land if the expression *becomes* true *or false*. Perhaps `ForceLandConditionChanged` would be best.

2. In the future `ConditionManager` will be `Variable Manager` along with other renaming with extended functionality, so `variables` would be preferred over `conditions` since I (or someone else) would have to rename it for consistency anyways.